Services/Concrete/CacheService.cs

```diff
@@ -108,6 +109,9 @@ public CacheService()
 
 		public async Task<List<Demo>> GetDemoListAsync()
 		{
+			if (_demoCache.Count > 0)
```

Comment:
If you change the stats date filter, the cache will return every time the same demos without filtering.
You should maybe add a parameter to force the cache to reload demos.
Or alternatively, loop through each demos in the cache and apply the filter on it.
If the cache doesn't contains any demos, just grab the data with GetDemoBasicDataAsync() as it is currently done.